controllers/space/space_controller_test.go

```diff
@@ -1029,6 +1031,127 @@ func TestUpdateSpaceTier(t *testing.T) {
 	})
 }
 
+// TestUpdateSpaceRoles covers the cases where SpaceBindings are created/updated/deleted,
+// and how this should affect the Space and its NSTemplateSet
+func TestUpdateSpaceRoles(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+
+	t.Run("add user with admin role", func(t *testing.T) {
+		// given a MUR, a Space and its NSTemplateSet resource...
+		mur := murtest.NewMasterUserRecord(t, "john")
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierName(basicTier.Name),
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity",
+			nstemplatetsettest.WithReferencesFor(basicTier),
+			nstemplatetsettest.WithReadyCondition())
+		// ...and a SpaceBinding for John as an Admin on the Space
+		sb := spacebindingtest.NewSpaceBinding(mur.Name, s.Name, "admin", "signupJohn")
+		hostClient := test.NewFakeClient(t, s, mur, sb, basicTier)
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, res.Requeue) // expect a requeue since the NSTemplateSet was updated
+		// Space should be in "updating" state while the
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, s.Name, hostClient).
+			HasConditions(spacetest.Updating())
+		// NSTemplateSet should have an spaceRoles entry for the `mur`
+		nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nstmplSet.Name, member1Client).
+			HasSpaceRoleForUser("basic-admin-123456new", mur.Name). // entry added for user
+			HasConditions(nstemplatetsettest.Provisioned())         // not changed by the SpaceController, but will be by the NSTemplateSetController
+	})
+
+	t.Run("remove user with admin role", func(t *testing.T) {
```

Comment:
could we modify the test case so in the `// given` part it would contain two space roles for one or two usernames? Then verify the removal of one SpaceBinding so in the `// then` part the `NSTemplateSet` would contain one SpaceRole/username less.

Reply:
done in 3c4927f, with 2 extra space bindings to verify that the existing entries remained unchanged.